opm/simulators/flow/FlowProblem.hpp

```diff
@@ -719,7 +718,28 @@ class FlowProblem : public GetPropType<TypeTag, Properties::BaseProblem>
     Scalar rockReferencePressure(const Context& context, unsigned spaceIdx, unsigned timeIdx) const
     {
         unsigned globalSpaceIdx = context.globalSpaceIndex(spaceIdx, timeIdx);
-        return this->rockReferencePressure(globalSpaceIdx);
+        return rockReferencePressure(globalSpaceIdx);
+    }
+
+    /*!
+    * \copydoc BlackoilProblem::rockReferencePressure
+    */
+    Scalar rockReferencePressure(unsigned globalSpaceIdx) const
+    {
+        const auto& rock_config = this->simulator().vanguard().eclState().getSimulationConfig().rock_config();
+        if (rock_config.store()) {
+            return asImp_().initialFluidState(globalSpaceIdx).pressure(refPressurePhaseIdx_());
+        }
+        else {
+            if (this->rockParams_.empty())
+                return 1e5;
```

Comment:
Can we please not use magic constants in our code?

Reply:
Everything in the `else {}` section here is from the original code previously located in `FlowGenericProblem_impl.hpp`, so I just moved it wholesale. From what I can see, just deleting the `if (this->rockParams_.empty())` clause will be OK since `rockReferencePressure()` will not be called without `ROCK` being present in the deck but I may be wrong in that assumption.

Reply:
I agree about reducing the number of magic constants in the code, but I would rather see that done in another PR.